src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -754,3 +758,79 @@ impl RustcTargetData {
         self.target_config(kind).links_overrides.get(lib_name)
     }
 }
+
+/// Structure used to deal with Rustdoc fingerprinting
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RustDocFingerprint {
+    _rustc_verbose_version: String,
+}
+
+impl RustDocFingerprint {
```

Comment:
I'm not sure it's really necessary to have a struct to wrap these methods.  From what I can see, this can all be done in a single free function, and it should be substantially smaller.

Reply:
I did it on that way so that if in the future more things need to be added to it, we already have the struct and it's easier to add functionalities. 

I can implement everything directly anyway if you prefer it. I was just thinking about the future, and more functionalities added to this fingerprinting feature for rustdoc.

What should I do then?

Reply:
I think if it grows in complexity in the future, it can be reworked to be more abstract, but for the foreseeable future I wouldn't expect it to change much.